OrbitCore/CallstackData.h

```diff
@@ -18,6 +18,26 @@ class CallstackData {
  public:
   explicit CallstackData() = default;
 
+  CallstackData(const CallstackData& other) = delete;
+
+  CallstackData& operator=(const CallstackData& other) = delete;
+
+  CallstackData(CallstackData&& other) {
+    absl::MutexLock lock(&(other.unique_callstacks_mutex_));
+    unique_callstacks_ = std::move(other.unique_callstacks_);
+    callstack_events_ = std::move(other.callstack_events_);
+  }
+
+  CallstackData& operator=(CallstackData&& other) {
+    absl::MutexLock lock(&(other.unique_callstacks_mutex_));
+    absl::MutexLock lock_other(&(other.unique_callstacks_mutex_));
```

Comment:
this can end up in a deadlock... In this sistuation we probably want to move all at once by moving unique_ptr of another (non-copiable/non-movable class with the mutex).

Reply:
Not sure, If I got you right? Should I remove the copy assignment operator and on the usage use unique_ptr and move that one?

Reply:
The possible deadlock is the reason why these operators weren't implemented before, and because of it we have std::unique_ptr<CallstackData> in CaptureData. But do you use these operators in this commit or why you want to add it?

Reply:
Initially, I wanted to get rid of passing the Pointer to CallstackData in particular to SamplingProfiler. However, as passing a non-copyable reference is not better, I've reverted the changes here and explicitly delete move and copy operators.